### DISCUSSION TITLE
Player stats by week?

### DISCUSSION THREAD

**Discussion by eggman920** [cwendt94/espn-api#556] [Q&A]
Since it's the offseason it's a bit tricky to test, but I wanted to see if there is a way to grab player stats by week. I'm currently using the following code but it is returning the full season stats. Lets say I wanted to grab all of the stats of each player per team roster in my league for week 13 is that possible? if possible, how could I modify the following code to do that?

```
df_list = []
for idx, i in enumerate(league.teams):
    team = league.teams[idx]
    print(team)
```

```
    for p in team.roster:
        print(p, p.position, p.stats)
        df = pd.json_normalize(p.stats[0]['breakdown'])
        df['team'] = team
        df['player'] = p
        df['pos'] = p.position
        df_list.append(df)
```

```
non_empty_dfs = [df for df in df_list if df is not None and not df.empty]
if not non_empty_dfs:
    print(f'Something went wrong')
else:
    combined_df = pd.concat(non_empty_dfs, ignore_index=True)
    print(combined_df)
```

**Comment by cwendt94** [comment]
To get player stats by week you will need to use the `player_info` call that gets all the player stats. The stats loaded at start up only have full season and projected season.

What you can do is loop through a `teams.roster` getting all the `playerIds` then calling `league.player_info(playerId=[...])` with all of those ids and then will return a list of Player Classes. Those stats should have every week stats.
